File: templates/_composition/MVVMLight/Feature.SuspendAndResume.AddNavigation/Services/SuspendAndResumeService_postaction.cs
using Microsoft.Practices.ServiceLocation;
namespace Param_ItemNamespace.Services
{
    internal class SuspendAndResumeService : ActivationHandler<LaunchActivatedEventArgs>
    {
        private async Task RestoreStateAsync()
        {
            //^^
            var navigationService = ServiceLocator.Current.GetInstance<NavigationServiceEx>();
            navigationService.Navigate(saveState.Target.FullName, saveState.SuspensionState);
        }
    }
}

File: templates/_composition/MVVMLight/Feature.SuspendAndResume.AddNavigationVB/Services/SuspendAndResumeService_postaction.vb
'{[{
Imports Microsoft.Practices.ServiceLocation
'}]}
Namespace Services
    Friend Class SuspendAndResumeService
        Inherits ActivationHandler(Of LaunchActivatedEventArgs)

        Private Async Function RestoreStateAsync() As Task
            '^^
            '{[{
            If saveState?.Target IsNot Nothing Then
                Dim navigationService = ServiceLocator.Current.GetInstance(Of NavigationServiceEx)()
                navigationService.Navigate(saveState.Target.FullName, saveState.SuspensionState)
            End If
            '}]}
        End Function
    End Class
End Namespace
